modules/openapi-generator/src/main/resources/dart2/auth/http_basic_auth.mustache

```diff
@@ -1,14 +1,14 @@
 {{>header}}
 {{>part_of}}
 class HttpBasicAuth implements Authentication {
-  HttpBasicAuth(this.username, this.password);
+  HttpBasicAuth({this.username, this.password});
 
-  String username;
-  String password;
+  String? username;
```

Comment:
I dont think making these nullable make any sense, they can't be null in HTTP spec, and we could get rid of null check below. Empty string also isn't a valid credential, you can't submit `agilob:` or `:password`. I think we should force users to provide non-nullable credentials and not do any checks on them.

Reply:
The code has a call to `HttpBasicAuth()` which will produce empty strings. How do we handle that?